Text file creation from API + DB connection to MySQL.

Cema2019 opened the pull request:
## Hi team,

I've refactored the original code to merge the fs module into the fruits.controller.js file to be able to export the API post entries into a text file. 
- At initial run, if no text file exists, it will get created. 
- Once created, it will feed the in-memory array for data persistence despite the server restarting. 
- If the file gets manually removed and the servers gets restarted, it will created the empty text file anew.

As new fruits get entered in the API, they will be written into the fruits.txt file using the fs.writeFile method. 
The API POST and GET functions have also been tested to ensure they display the current data. 

![image](https://github.com/user-attachments/assets/dba98549-0fd4-4639-ba0a-22dc36f7a09a)

![image](https://github.com/user-attachments/assets/2bfb9a17-3d1e-4b59-b65a-a28818b80ec8)

![image](https://github.com/user-attachments/assets/f41166fc-aa83-4ef5-89be-e0d04912f353)

Ralexs0096 committed:
Access to the query and params from request

Ralexs0096 committed:
Add dev script

Ralexs0096 committed:
Relocate endpoints functions to controllers directory

Ralexs0096 committed:
Add nodemon dependency

Ralexs0096 committed:
update dev command

Ralexs0096 committed:
Add new example using FS api

Cema2019 committed:
Edit fs methods

Cema2019 committed:
Add validation to prevent repeated fruits

Cema2019 committed:
Update logic to import fs methods to export to txt file

Cema2019 committed:
Refactor code to change fs.exists to fs.access due to its deprecation

Cema2019 committed:
Create db connection to MySQL

Cema2019 committed:
Add printFruitsFromDb function to connect to DB and print its data

Cema2019 committed:
Update package.json file

Cema2019 committed:
Update printFruitsFromDb function after successful testing

Cema2019 commented:
Hi team,
I updated the PR to include the connection to the MySQL database from Node.js

![image](https://github.com/user-attachments/assets/c35707c9-3d4f-428b-93f1-3a9ea330120a)
Successful connection
![image](https://github.com/user-attachments/assets/78707aa4-4433-40b8-b755-9228dee110cb)
Failed connection
![image](https://github.com/user-attachments/assets/254517a0-4b88-4580-b0c8-c71ab698073f)

Ralexs0096 commented on src/controllers/fruits.controller.js:
```diff
@@ -9,13 +68,26 @@ export const getAllFruits = (_, response) => {
-    // Add validation to prevent repeated fruits
-    fruits.push(fruit);
+  if (!fruit) {
+    return res.status(400).send({
+      message: 'Fruit name is required',
+      ok: false
+    });
```
```suggestion
    res.status(400).send({
      message: 'Fruit name is required',
      ok: false
    });
    return
```

Cema2019 replied:
done!

Cema2019 committed:
Updated createAFruit to match suggested change from PR

Cema2019 committed:
Create SQL queries

Cema2019 committed:
Refactor file. Remove fs functions and implement assigments for db queries

Cema2019 committed:
Update controller function names to index.js

Cema2019 commented:
Updated branch to include controller function getFruitByIdQuery and deleteFruitById.
Also removed references to fs coding.
Below tests performed with RapidAPI.

![image](https://github.com/user-attachments/assets/9e969c04-0311-4d23-8418-f0b76fe4a6a0)
![image](https://github.com/user-attachments/assets/0d90cf90-e5f0-4fc7-afba-1e3212a2a099)
![image](https://github.com/user-attachments/assets/ae859653-f3fc-4513-a77c-8f1f764b25c4)
![image](https://github.com/user-attachments/assets/0dd29af5-07e9-4abe-b167-12faa3619f10)
![image](https://github.com/user-attachments/assets/225134ed-dd21-4132-9f38-3d0190986ca9)
![image](https://github.com/user-attachments/assets/a2349d5d-7363-4187-b614-da7616dfd83c)

Cema2019 committed:
Configure Delete endpoint

Cema2019 committed:
Import knex and dotenv dependencies

Cema2019 committed:
Create .env file to save credentials

Cema2019 committed:
Refactor database.js to knex + dotenv configuration

Cema2019 committed:
Remove queries folder. Logic will be provided in the controllers file

Cema2019 committed:
Refactor file to to use Knex and load credentials from environment variables.

Cema2019 commented:
Hi team,

Updated branch to refactor coding with knex.js (instead of raw SQL) and dotenv libraries to better keep credentials.
Below tests performed with RapidAPI.

![image](https://github.com/user-attachments/assets/9dff7f79-f0d1-4c63-bcbe-beb8cbba9e22)

![image](https://github.com/user-attachments/assets/861c7cec-0bf0-475c-bf8c-d1547a723071)

![image](https://github.com/user-attachments/assets/ffa23d20-5b9d-4c89-9288-93b1a7070960)

![image](https://github.com/user-attachments/assets/06a41d17-4af6-4b0a-9faf-80bd62bfa92c)

![image](https://github.com/user-attachments/assets/9e410b43-f94c-49b8-9757-ca8e06d64c0c)

![image](https://github.com/user-attachments/assets/8408914b-25ff-420e-8ed9-50e7ca7b39ce)

Cema2019 committed:
Create new tables on MySQL Workbench for users and posts. Also create new controller file for users

Cema2019 committed:
Work on coding for post controller file

Cema2019 committed:
Edit controllers for users and post

Cema2019 commented:
![image](https://github.com/user-attachments/assets/f60610a1-b3ba-4e52-9c4f-25d4e180b31b)

![image](https://github.com/user-attachments/assets/b5d887ce-63fe-485b-a9d8-c1d4e5503828)

![image](https://github.com/user-attachments/assets/dfc49d2b-f22d-4b5e-bc68-77a1ce315539)

![image](https://github.com/user-attachments/assets/c32a3874-26bf-4ab9-80a7-d45a321de37b)

![image](https://github.com/user-attachments/assets/c054ab4e-30b6-488f-9219-6a710cfd3450)

![image](https://github.com/user-attachments/assets/fc74a22e-8a3d-4f00-9513-f408fb44cf71)

Cema2019 committed:
Update variable naming

Cema2019 committed:
Update server routes to plural to align with RESTful conventions

Cema2019 committed:
Perform code refactoring

Cema2019 committed:
Create UpdateFruit function for PUT method

Cema2019 committed:
Rename functions for consistency
